pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/DictionaryBasedAggregationOperator.java

```diff
@@ -74,15 +76,27 @@ protected IntermediateResultsBlock getNextBlock() {
           AggregationFunctionFactory.AggregationFunctionType.valueOf(function.getName().toUpperCase());
       String column = aggregationFunctionContext.getAggregationColumns()[0];
       Dictionary dictionary = _dictionaryMap.get(column);
-      AggregationResultHolder resultHolder;
+      FieldSpec.DataType dataType = _dataTypeMap.get(column);
+      AggregationResultHolder resultHolder = AggregationResultHolderFactory.getAggregationResultHolder(functionType, dataType);
       switch (functionType) {
       case MAX:
-        resultHolder = new DoubleAggregationResultHolder(dictionary.getDoubleValue(dictionary.length() - 1));
+        if (dataType.equals(FieldSpec.DataType.STRING)) {
```

Comment:
Minor nit: I prefer having the 'if' around reading the value from dictionary, and having resultHolder.setValue() outside of the 'if'.